MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/AnalyticsController.java

```diff
@@ -1,44 +1,96 @@
+        this.analyticsService = analyticsService;
+    }
 
-    private UserService userService;
-    private RestTemplate restTemplate;
+    /**
+     * Handles requests for getting list of all user's products
+     *
+     * @param session - HttpSession object
+     * @param model   - a map that will be handed off to the view for rendering the data to the client
+     * @return - string with appropriate view name
+     */
+    @GetMapping()
+    public String getProducts(HttpSession session, Model model) {
+        ProductStatistics productStatistics = (ProductStatistics) session.getAttribute("productStatistics");
+        if (productStatistics != null) {
+            model.addAttribute("productStatistics", productStatistics);
+            return "analytics";
+        }
+        List<AnalyticsProductDTO> analyticsProductDTOs = analyticsService.getProducts();
 
-    @Autowired
-    public AnalyticsController(UserService userService, RestTemplate restTemplate) {
-        this.userService = userService;
-        this.restTemplate = restTemplate;
+        model.addAttribute("analyticsProductDTOs", analyticsProductDTOs);
+
+        return "selectProductForAnalytics";
     }
 
     /**
-     * This method returns a analytics view.
+     * Handles requests for getting analytics about concrete product
      *
-     * @return analytics view url
+     * @param model     - a map that will be handed off to the view for rendering the data to the client
+     * @param productId - product unique identifier
+     * @return string with appropriate view name
      */
-    @GetMapping
-    public String getAnalyticsPage() {
-        restTemplate.getForEntity(String.format(restAnalyticsURL, userService.getCurrentUser().getId()), String.class);
+    @GetMapping("/{productId}")
+    public String getProductStatistics(HttpSession session, Model model, @PathVariable int productId) {
+        ProductStatistics productStatistics = analyticsService.getProductStatistics(productId);
+        model.addAttribute("productStatistics", productStatistics);
+        session.setAttribute("productStatistics", productStatistics);
+
         return "analytics";
     }
 
+    /**
+     * Handles requests for clean ProductStatistics session object
+     *
+     * @return string with appropriate view name
+     */
+    @GetMapping("/cleanSession")
+    public String cleanProductStatisticsSessionObject() {
+        analyticsService.cleanProductStatisticsSessionObject();
+
+        return "redirect:/analytics";
+    }
+
+    /**
+     * Handles HttpClientErrorException exceptions
+     *
+     * @param model - a map that will be handed off to the view for rendering the data to the client
+     * @param e     - HttpClientErrorException exception
+     * @return string with appropriate view name
+     */
+    @ExceptionHandler(HttpClientErrorException.class)
+    public String handleRestClientException(Model model, HttpClientErrorException e) {
+        {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Extract this nested code block into a method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1199)